src/github.com/getlantern/peerscanner/web.go

```diff
@@ -149,12 +149,29 @@ func clientIpFor(req *http.Request, name string) string {
 	ip := strings.TrimSpace(ips[0])
 	// TODO: need a more robust way to determine when a non-fallback host looks
 	// like a fallback.
-	if !isFallback(name) && strings.HasPrefix(ip, "128.199") {
+	hasFallbackIp := isFallbackIp(ip)
+	if !isFallback(name) && hasFallbackIp {
 		log.Errorf("Found fallback ip %v for non-fallback host %v", ip, name)
 		return ""
-	} else if isFallback(name) && !strings.HasPrefix(ip, "128.199") {
+	} else if isFallback(name) && !hasFallbackIp {
 		log.Errorf("Found non-fallback ip %v for fallback host %v", ip, name)
 		return ""
 	}
 	return ip
 }
+
+func isFallbackIp(ip string) bool {
+	for _, prefix := range fallbackIPPrefixes {
+		if strings.HasPrefix(ip, prefix) {
+			return true
+		}
+	}
+
+	return false
+}
+
+var fallbackIPPrefixes = []string{
+	"128.199",
+	"178.62",
+	"188.166",
+}
```

Comment:
> I don't understand. In this scenario I'd imagine the server name is more reliable than the IP? How would the fallback in the middle get to mess with the name?

Sorry, there are two scenarios:

1. Peer registering through fallback and accidentally getting its IP
2. Malicious peer registering itself with a non-fallback IP but a fallback name (and thus getting into the fallbacks rotation)

I'm attempting to protect against both, but the protection for no. 2 is fairly week since someone could set up a peer that has a fallback-like IP (for example by putting it on DO).